rskj-core/src/test/java/co/rsk/peg/federation/FederationSupportImplTest.java

```diff
@@ -2191,6 +2195,50 @@ void clearRetiredFederation_whenHavingOldFederation_removesOldFederation() {
         assertThat(oldFederation, is(nullValue()));
     }
 
+    @Test
+    @Tag("create and process svp fund transaction unsigned")
+    void createAndProcessSvpFundTransactionWithoutSignatures_savesExpectedTransactionInStorage() throws InsufficientMoneyException {
+        // arrange
+        BridgeStorageProvider bridgeStorageProvider = mock(BridgeStorageProvider.class);
+
+        Federation proposedFederation = new P2shErpFederationBuilder().build();
+        storageProvider.setProposedFederation(proposedFederation);
+
+        Coin spendableValueFromProposedFederation = BridgeMainNetConstants.getInstance().getMinimumPegoutTxValue();
+        Coin feePerKb = Coin.valueOf(1000);
+
+        // create the active federation wallet with utxos
+        Federation activeFederation = federationSupport.getActiveFederation();
+        List<UTXO> utxos = BitcoinTestUtils.createUTXOs(10, activeFederation.getAddress());
+        Wallet activeFederationWallet =
+            BridgeUtils.getFederationSpendWallet(new Context(mainnetParams), activeFederation, utxos, true, bridgeStorageProvider);
+
+        // act
+        BtcTransaction svpFundTransactionUnsigned = federationSupport.createAndSetSvpFundTransactionWithoutSignatures(spendableValueFromProposedFederation, feePerKb, activeFederationWallet);
+
+        // assertions
+        // assert hash was saved in storage entry
+        Optional<Sha256Hash> svpFundTransactionHashUnsignedOpt = storageProvider.getSvpFundTxHashUnsigned(allActivations);
+        assertTrue(svpFundTransactionHashUnsignedOpt.isPresent());
+
+        List<TransactionOutput> svpFundTransactionUnsignedOutputs = svpFundTransactionUnsigned.getOutputs();
+        // assert one of the outputs was sent to the proposed federation
+        Script proposedFederationScriptPubKey = proposedFederation.getP2SHScript();
+        Optional<TransactionOutput> outputToProposedFederationOpt = svpFundTransactionUnsignedOutputs.stream()
+            .filter(output -> output.getScriptPubKey().equals(proposedFederationScriptPubKey))
+            .findFirst();
+        assertTrue(outputToProposedFederationOpt.isPresent());
+        // assert the amount sent was correct
+        TransactionOutput outputToProposedFederation = outputToProposedFederationOpt.get();
+        assertEquals(outputToProposedFederation.getValue(), spendableValueFromProposedFederation);
+
+        // assert all the other outputs are the change sent to the active federation
+        Script activeFederationScriptPubKey = activeFederation.getP2SHScript();
+        assertTrue(svpFundTransactionUnsignedOutputs.stream()
+            .filter(output -> !output.equals(outputToProposedFederation))
+            .allMatch(output -> output.getScriptPubKey().equals(activeFederationScriptPubKey)));
+    }
```

Comment:
assert the transaction is added to the pegout tx index?

Reply:
isn't that [this assertion](https://github.com/rsksmart/rskj/pull/2677/files#diff-bca5614b0a0ce7ba81d79322090b66cee8d9289e3430798d67767568abc8491aR637-R641)?